### DISCUSSION TITLE
apollo code:gen to generate types, queries, mutation subscription etc

### DISCUSSION THREAD

**Discussion by mfahad017** [graphql/graphql-js#3664] [Q&A] [answered]
I am using the latest syntax I guess where you write types like this

`const LoginType = new GraphQLObjectType({
  name: "Login",
  fields: () => ({
    user: { type: UserType.UserType },
    tokens: { type: TokensType },
  }),
});
const LoginInputType = new GraphQLInputObjectType({
  name: "LoginInputType",
  fields: () => ({
    email: { type: new GraphQLNonNull(GraphQLString) },
    password: { type: new GraphQLNonNull(GraphQLString) },
  }),
});`

My BE is working fine. Now I want to generate type, queries etc on FE I looked into an article and other all of them use the other approch to create gql types in string

**Comment by saihaj** [comment, marked as answer]
1. This is not Apollo codegen so questions should be asked in their project.
2. FWIW GraphQL Schema is used to generate types from schema.
3. This question is too vague and would recommend reading more about queries.
4. I would encourage asking this question in community https://graphql.org/community/
